packages/protocol/contracts-0.8/common/EpochManager.sol

```diff
@@ -319,4 +336,42 @@ contract EpochManager is
       CELOequivalent
     );
   }
+
+  /**
+   * @notice Sends the allocated epoch payment to a validator, their group, and
+   *   delegation beneficiary.
+   * @param validator Account of the validator.
+   */
+  function sendValidatorPayment(address validator) external {
+    IAccounts accounts = IAccounts(getAccounts());
+    address signer = accounts.getValidatorSigner(validator);
+
+    FixidityLib.Fraction memory totalPayment = FixidityLib.newFixed(
+      validatorPendingPayments[signer]
+    );
+
+    IValidators validators = getValidators();
+    address group = validators.getValidatorsGroup(validator);
+    (, uint256 commissionUnwrapped, , , , , ) = validators.getValidatorGroup(group);
+
+    uint256 groupPayment = totalPayment.multiply(FixidityLib.wrap(commissionUnwrapped)).fromFixed();
+    FixidityLib.Fraction memory remainingPayment = FixidityLib.newFixed(
+      totalPayment.fromFixed() - groupPayment
+    );
+    (address beneficiary, uint256 fraction) = getAccounts().getPaymentDelegation(validator);
+    uint256 delegatedPayment = remainingPayment.multiply(FixidityLib.wrap(fraction)).fromFixed();
+    uint256 validatorPayment = remainingPayment.fromFixed() - delegatedPayment;
+
+    IStableToken stableToken = IStableToken(getStableToken());
+
+    require(stableToken.transfer(validator, validatorPayment), "mint failed to validator");
```

Comment:
not mint but transfer. Can transfer fail?

Reply:
is validatorPayment always > 0?

Reply:
> not mint but transfer. Can transfer fail?

If the contract doesn't have enough balance for example. Ideally that should never happen though, if our allocation and minting logic is correct. Auditors in the past have asked to always check return values of e.g. ERC-20 functions that return a success `bool`, as good practice.

> is validatorPayment always > 0?

I guess it could be 0, if delegation and/or commission take up the whole payment. I can add an `if(non-zero)` around this as well. That said, transfers of 0 do not fail, as evidenced by the `test_doesNothingIfNotAllocated` test not reverting.